src/NosCore.GameObject/Providers/MinilandProvider/MinilandProvider.cs

```diff
@@ -155,9 +157,20 @@ public void SetState(long characterId, MinilandState state)
                 var ml = _minilandIds[characterId];
                 var miniland = _mapInstanceProvider.GetMapInstance(ml.MapInstanceId);
                 ml.State = state;
+
                 if (ml.State != MinilandState.Open)
                 {
-                    miniland.Kick(o => o.VisualId != characterId);
+                    if (ml.State == MinilandState.Private)
+                    {
+                        List<long> friends = _friendHttpClient.GetListFriends(characterId)
+                            .Select(s => s.CharacterId)
+                            .ToList();
+                        // Kick all players in miniland except owner and his friends
```

Comment:
this comment is not really useful. try having comments only when really nescessary